middlewarex/oidc.go

```diff
@@ -0,0 +1,100 @@
+package middlewarex
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+)
+
+type (
+	OIDCMiddlewareOption func(*oidcMiddleware)
+
+	oidcMiddleware struct {
+		authFailedHandler func(error) http.HandlerFunc
+		config            *oidc.Config
+		verifier          *oidc.IDTokenVerifier
+	}
+	oidcContextKey int
+)
+
+const idTokenContextKey oidcContextKey = iota
+
+func OIDC(ctx context.Context, issuer string, opts ...OIDCMiddlewareOption) func(next http.Handler) http.Handler {
+	provider, err := oidc.NewProvider(ctx, issuer)
+	if err != nil {
+		panic(err)
```

Comment:
Should we panic here or do it like
https://github.com/HGV/iam-api/commit/870634b0f39e3794edc49b43abf0dac9578ecbde

Reply:
I would either panic or return the middleware as a struct and an error in this case.

Reply:
According to https://go.dev/doc/effective_go#panic it’s acceptable to `panic()` in this case because the OIDC middleware can’t set itself up. But this could also be seen from another perspective: What if it isn’t a critical error for the app if the OIDC middleware fails. For example, it could fall back to another authentication mechanism. Just hypothesizing.  In that case it would be better to return the `error` during setup instead of `panic()`. What’s your take on this?